src/server/page_server.cpp

```diff
@@ -224,10 +224,14 @@ page_server::tran_server_connection_handler::receive_start_catch_up (tran_server
   _er_log_debug (ARG_FILE_LINE,
 		 "[CATCH_UP] It's been requested to start catch-up with the follower (%s:%d), until LSA = (%lld|%d)\n",
 		 host.c_str (), port,LSA_AS_ARGS (&catchup_lsa));
+
   if (port == -1)
     {
       // TODO: It means that the ATS is booting up.
       // it will be set properly after ATS recovery is implemented.
+
+      // TODO reply the catchup_done msg.
+      log_Gl.get_log_prior_receiver ().start_thread ();
```

Comment:
Nit.
I think that we must first start the thread and then reply with the catch-up done message - to avoid race conditions:
- PS returns the message
- ATS starts pushing already cached messages
- PS log prior receiver thread is not started

It practice it is probably impossible for this to happen.

Reply:
It's not the case because the `start_thread()` just starts consuming the log records and all log records from ATS may be already accumulated in the receiver's queue.

Regarding this, it could be more readable to change the name of the function from `start_thread()` to `start_consumer_thread()`. What do you think?

Reply:
> It's not the case because the start_thread() just starts consuming the log records and all log records from ATS may be already accumulated in the receiver's queue. So, the order doesn't count.

It is my understanding that the PS does not receive the log prior messages until the catch-up is done. But I think I'm wrong because in `active_tran_server::connection_handler::on_connecting` the log prior sink si registered before the catch-up process. So, my bad.

`log_Gl.get_log_prior_receiver ().start_thread ();` is enough for me, as it is already topological: prior received only contains a single thread.